File: codetraverse/registry/extractor_registry.py
from codetraverse.extractors.haskell_extractor import HaskellComponentExtractor
from codetraverse.extractors.python_extractor import PythonComponentExtractor
from codetraverse.extractors.rescript_extractor import RescriptComponentExtractor
from codetraverse.extractors.rust_extractor import RustComponentExtractor
from codetraverse.extractors.go_extractor import GoComponentExtractor
from codetraverse.extractors.typescript_extractor import TypeScriptComponentExtractor
from codetraverse.extractors.purescript_extractor import PureScriptComponentExtractor
from codetraverse.extractors.javascript_extractor import JavascriptExtractor

def get_extractor(language: str):
    lang = language.lower()
    if lang == "haskell":
        return HaskellComponentExtractor()
    if lang == "python":
        return PythonComponentExtractor()
    if lang == "rescript":
        return RescriptComponentExtractor()
    if lang == "rust":
        return RustComponentExtractor()
    if lang == "golang":
        return GoComponentExtractor()
    if lang == "typescript":
        return TypeScriptComponentExtractor()
    if lang == "purescript":
        return PureScriptComponentExtractor()
    if lang == "javascript":
        return JavascriptExtractor()
    raise ValueError(f"No extractor for language: {language}")

File: codetraverse/registry/extractor_registry.ts
import { HaskellComponentExtractor } from '../extractors/haskell_extractor';
import { PythonComponentExtractor } from '../extractors/python_extractor';
import { RustComponentExtractor } from '../extractors/rust_extractor';
import { TypeScriptComponentExtractor } from '../extractors/typescript_extractor';
// Import other extractors here

const extractorMap: Record<string, any> = {
    "haskell": HaskellComponentExtractor,
    "python": PythonComponentExtractor,
    // "rescript": RescriptComponentExtractor,
    "rust": RustComponentExtractor,
    // "golang": GoComponentExtractor,
    "typescript": TypeScriptComponentExtractor,
    // "purescript": PurescriptComponentExtractor,
    // "javascript": JavascriptComponentExtractor
};

export function getExtractor(language: string) {
    const extractor = extractorMap[language];
    if (!extractor) {
        throw new Error(`No extractor found for language: ${language}`);
    }
    return new extractor();
}
